elastic/kibana
# [Security Solution] [Elastic AI Assistant] Fix Send to Timeline ESQL action changing the ESQL bar to KQL bar

## Summary

In https://github.com/elastic/kibana/pull/169478, the `Send to Timeline` action was fixed for `ES|QL` codeblocks, however while the `ES|QL` timeline tab was being selected, and the query string was being set correctly, the ESQL bar would be swapped out for the KQL bar. In the gifs below, note the right two action buttons present in the query bar.

There is one remaining bug where refreshing the page clears the ESQL query and set the default query. @logeekal is currently investigating a fix for this issue and will push to this branch once available.

#### Before
<p align="center">
  <img width="500" src="https://github.com/elastic/kibana/assets/2946766/fdc3b2a0-5b4b-4584-b304-c4d24de1917c" />
</p> 

#### After
<p align="center">
  <img width="500" src="https://github.com/elastic/kibana/assets/2946766/d217b95c-2d1c-4474-ac98-180e07a86669" />
</p> 


## Test instructions

Either request the assistant to generate an ESQL query or just paste this codeblock into the conversation to test the action directly from the user message. Be sure to declare the codeblock language as `esql` or include one of the [string match patterns](https://github.com/elastic/kibana/pull/169478/files#diff-f70f0b96568e024e53bfbb62adcca72051f0a2e824d4ab22664eed0e149be248R38) above the code block so the action can be recognized.


````
Below is an `Elasticsearch Query Language` query:

```esql
FROM logs-endpoint*
| WHERE event.category == \"process\"
| STATS proc_count = COUNT(process.name) BY host.name
| KEEP host.name, proc_count
```
````


Then verify that the query bar is still the ESQL query bar which has the `Expand` and `ESQL Reference` buttons to the far right.
